### ISSUE TITLE
Double Dipping on Stat Bonuses

### ISSUE THREAD

**Issue by dmlfan928** [OpenRCT2/OpenRCT2#20146]
### Operating System

Windows 10 64-Bit

### OpenRCT2 build

OpenRCT2, v0.4.4-230-g61d820e (61d820e on develop) provided by GitHub

### Base game

RollerCoaster Tycoon 2

### Describe the issue

I noticed that the game now calculates stat bonuses for choosing the new "reverse train" option. This option now allows you to double dip on stat bonuses with coasters that have a reversed train option in the base game. What this leads to is a normal forwards facing train getting the reversed bonuses twice. Attached is an example of a simple "Classic Wooden Coaster."  
The first picture is the base stats. 
2nd is with the regular trains with the reversed option selected. 
3rd is with the reverse trains available in the base game without the Open RCT2 reversed option selected. 
4th is the base game reversed trains with the Open RCT2 reversed option selected. 

### Steps to reproduce

1. Build a ride with reversed trains available in the base game
2. use the base game reversed trains
3. use the Open RCT2 reverse trains option and the bonuses will be added twice.

### Attachments

![20230507_090131](https://user-images.githubusercontent.com/95002694/236679448-4b21ef5f-bd36-482c-8cef-a7ed2152c1eb.jpg)
![20230507_090144](https://user-images.githubusercontent.com/95002694/236679449-fa1d31a5-2ae1-418e-96bd-a07b03a3072f.jpg)
![20230507_090221](https://user-images.githubusercontent.com/95002694/236679451-0f830a8c-540a-4679-bd29-92ef1945435f.jpg)
![20230507_090247](https://user-images.githubusercontent.com/95002694/236679453-5a8d07ed-3d87-4995-bb3d-db9b273c7f31.jpg)

**Comment by karst** [comment]
AFAIK the idea is to merge the reverse trains together with the regular trains afaik. Convert base game reversed trains into regular trains, with the reverse option selected, when opening a park in the new version. This is yet to be done, though.

@ZeeMaji can you confirm?

**Comment by ZeeMaji** [comment]
I believe that is the plan as it was discussed in the discord a while back, as it does seem rather odd to keep the separate reverse objects around now with the reversed trains option properly emulating the original rating multipliers.
However @kyphii would probably have better insight on this seeing as they were the one who implemented this whole new reversed trains system.

**Comment by kyphii** [comment]
Yes this is the plan, I am going to work on phasing out the reversed train types, along with the reversed vehicle subtypes for the Heartline Coaster and Log Flume. This stat exploit will exist for a little while though as I wasn't able to get it in in time for this release.

**Comment by MASHzj** [comment]
Question: In many scenarios the reversed train types are unlocked separately. So what would happen to the research process if those trains will be phased out?

**Comment by ZeeMaji** [comment]
I don't believe that's been discussed, but i would assume they will just be simply removed from the invention lists when importing old s4/s6/.park files?
Although there could be weird custom scenarios out there where you research the reversed trains before the normal ones (or not even have the normal ones at all!) so maybe in those cases it would just change the reversed trains to the normal trains in the inventions list? (and enable the normal trains object if it's not enabled)

**Comment by HtotheTML** [comment]
I take it some compatibility issues would exist, then?
